Java: Type cache (24.12) [STUD-72718]

viogroza opened the pull request:
Change target framework from 5.0 to 6.0
Add view models project
Add icon resource
Fix activity converter

lucarachiteanu commented on Activities/Java/UiPath.Java.Activities.ViewModels/UiPath.Java.Activities.ViewModels.csproj:
```diff
@@ -0,0 +1,27 @@
+
+  <ItemGroup>
+    <EmbeddedResource Include="Resources\ActivitiesMetadataWindows.json" />
+    <EmbeddedResource Include="Resources\Icons\JavaIcon.svg" />
```
```suggestion
    <EmbeddedResource Include="Resources\Icons\*.svg" />
```

viogroza replied:
will do

lucarachiteanu commented on Activities/Java/UiPath.Java.Activities.ViewModels/UiPath.Java.Activities.ViewModels.csproj:
```diff
@@ -0,0 +1,27 @@
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="Resources\ActivitiesMetadataWindows.json" />
```
```suggestion
    <EmbeddedResource Include="Resources\ActivitiesMetadataWindows.json" LogicalName="$(RootNamespace).Resources.ActivitiesMetadata.json" />
```

Since we do not target net6-windows for activities, it is unclear to me as to how do we discern between Cross and Windows?

viogroza replied:
the change you pointed out is necessary
at the moment the activities are available on only on windows (even if java.activities is a cross-platform project and java in general is cross-platform)

lucarachiteanu commented on Activities/Java/UiPath.Java.Activities.ViewModels/Resources/ActivitiesMetadataWindows.json:
```diff
@@ -0,0 +1,48 @@
+{
+  "orderedCategoryDisplayNameKeys": [],
```
```suggestion
  "activities": [],

```

I think this is intended for property level metadata when listing the properties of an activity.

viogroza replied:
I will remove it

viogroza committed:
Java: Type cache (24.12) [STUD-72718]

Change target framework from 5.0 to 6.0
Add view models project
Add icon resource
Fix activity converter
Fix pipeline to use NET 6.0
Address review findings
